src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs

```diff
@@ -187,14 +194,23 @@ private object TransformInput(object input, Dictionary<string, object> bindingDa
             return input;
         }
 
-        private CancellationToken HandleCancellationTokenParameter(object input)
+        private CancellationToken HandleCancellationTokenParameter(object input, HttpRequest httpRequest = null)
         {
-            if (input == null)
+            if (input is null) // TODO: if input is null but RequestAborted is not, we should still use that CT
             {
                 return CancellationToken.None;
             }
 
-            return (CancellationToken)input;
+            CancellationToken cancellationToken = (CancellationToken)input;
+
+            if (httpRequest is not null)
+            {
+                // TODO: dispose cancellationSource
+                var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, httpRequest.HttpContext.RequestAborted);
```

Comment:
Is this the best approach for allowing cancellation for both the host cancellation token and the http one? 

Suggestions for disposal?